### DISCUSSION TITLE
Riya'nın doğum günü

### DISCUSSION THREAD

**Discussion by Berkek007** [sonsuzus/sonsuzus#372] [Soru Cevap]
Problem
 
Madhav Riya'nın Doğum Günü Partisine gitti. O bir inek, bu yüzden hangi hediye hoşuna gideceği konusunda hiçbir fikri yoktu. Bu yüzden bir dizi tamsayıyı yanına aldı. Dizi belirli bir düzene uyuyordu. Dizinin ilk elemanı 1'di. Dizinin ikinci elemanı 6 idi.
Dizinin diğer elemanları, kendisinden önce gelen ve sonrasındaki sayının aritmetik ortalamasının iki eksiği olarak belirleniyordu. Açıkça görüleceği gibi, Riya bu fikrin aptalca olduğunu düşündü ve bu yüzden Madhav'ı cezalandırmak istedi. Madhav'ın bu utanç verici durumdan kaçmasına yardım et.

Girdi

Girdi, T, Test Durumlarının sayısı ile başlar.
Sonraki T satırda tamsayı N bulunur.

Çıktı

Her test durumu için, dizinin N'inci bir tamsayısını çıktı olarak verin. Cevap çok büyük olabileceğinden, sonucu 10 üzeri 9 + 7'e göre modulo alarak çıktı verin.

Kısıtlamalar:

1 ≤ T ≤ 105
1 ≤ N ≤ 1018

Örnek Giriş
2
1
3
Örnek Çıktı
1
15

Soru kaynağı: https://www.hackerearth.com/practice/math/number-theory/basic-number-theory-1/practice-problems/algorithm/riyas-birthday-party-1/

**Comment by nilbuyukkinaci** [comment]
```c
#include <iostream>
#include <string>
using namespace std;

int dizi[1023], t, n[110];

int main() {
    dizi[0] = 1;
    dizi[1] = 6;

    for (int i = 2; i < 1018; i++)
        dizi[i] = (dizi[i - 1] + 2) * 2 - dizi[i - 2];

    cin >> t;
    
    for (int i = 0; i < t; i++)
        cin >> n[i];
    
    for(int i = 0; i < t; i++)
        cout << dizi[n[i] - 1] << endl;
}
```

**Reply by sonsuzus** [reply]
Runtime error alıyor kodun.

**Reply by sonsuzus** [reply]
Doğru sonuçlar üretiyor da muhtemelen girilen sayılar dizi boyutlarını aşıyor.

**Comment by sonsuzus** [comment]
İpucu olsun diye işe yarar bir kod ekliyorum ama bu kod testleri geçmiyor haberiniz olsun. (Doğru sonucu verse de zaman ve bellek durumuna takılıyor)

```python
dizi = [1,6]
N = []
T = int(input())
for _ in range(T):
    N.append(int(input()))

ma = max(N)
for i in range(2,ma):
    dizi.append(2*dizi[i-1]-dizi[i-2]+4)

for i in N:
    print(dizi[i-1])
```

**Comment by Murkrow067** [comment]
```c
#include <stdio.h>

int main() 
{
    int x, dizi[100000] = {1,6};
    for (int i = 2; i < 100000; i++)
    {
        dizi[i] = (dizi[i - 1] + 2) * 2 - dizi[i - 2];
    }
    printf("Hangi siradaki sayiyi gormek istersiniz? \n");
    scanf("%d",&x);
    printf("%d. siradaki sayi: %d",x,dizi[x-1]);
}
```
